### PR TITLE
store: make marshalling internal

### PR DESCRIPTION
**Pull request by teg**
Make the types used for marshalling internal to the store package to reduced the interdependencies between our packages. In follow-ups we will simplify the store API further by exposing real types/objects rather than ones meant for marshalling.

### CODE REVIEW

File: internal/store/fixtures.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/common"
+	"github.com/osbuild/osbuild-composer/internal/compose"
+	"github.com/osbuild/osbuild-composer/internal/target"
+)
+
+func FixtureBase() *Store {
```
Comment (larskarlitski) [inline internal/store/fixtures.go:13]: It's a bit odd to have the fixtures be public types on the store. Is the idea to make them private at some point?

Comment (teg) [reply internal/store/fixtures.go:13]: Not sure about that. I think keeping them public is in keeping with the way the fixtures are meant to be used, but I'm happy to discuss that in a follow-up, unless you think it is a blocker?

File: internal/weldr/api_test.go [hunk 2]
```diff
@@ -432,6 +433,10 @@ func TestBlueprintsDepsolve(t *testing.T) {
 }
 
 func TestCompose(t *testing.T) {
+	arch, err := test_distro.New().GetArch("x86_64")
+	assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (teg) [reply internal/weldr/api_test.go:438]: Agreed. Fixed!

File: internal/store/json.go [hunk 3]
```diff
@@ -86,20 +87,28 @@ func newWorkspaceFromV0(workspaceStruct workspaceV0) map[string]blueprint.Bluepr
...
 		c := Compose{
 			Blueprint: composeStruct.Blueprint,
 		}
-		if len(composeStruct.ImageBuilds) == 0 {
-			panic("the was a compose with zero image builds, that is forbidden")
-		}
 		for _, imgBuild := range composeStruct.ImageBuilds {
+			imgTypeString, valid := imgBuild.ImageType.ToCompatString()
+			if !valid {
+				// Invalid type strings in serialization format, this may happen
```
[the remark this answers is not in the document]
Comment (teg) [reply internal/store/json.go]: Yeah, you are right. Added.

File: internal/test/helpers.go [hunk 4]
```diff
@@ -157,6 +158,14 @@ func Ignore(what string) cmp.Option {
 	return cmp.FilterPath(func(p cmp.Path) bool { return p.String() == what }, cmp.Ignore())
 }
 
+func CompareImageTypes() cmp.Option {
```
Comment (msehnout) [inline internal/test/helpers.go:161]: What is the idea behind this function? Although the code seems reasonable, it's not obvious why only these 3 properties were selected for comparison as opposed to e.g. MIMEType or ModulePlatformID. Maybe add a comment explaining it?

File: internal/store/json.go [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+func newComposesFromV0(composesStruct composesV0) map[uuid.UUID]compose.Compose {
+	composes := make(map[uuid.UUID]compose.Compose)
+
+	for composeID, composeStruct := range composesStruct {
+		c := compose.Compose{
+			Blueprint: composeStruct.Blueprint,
+		}
+		if len(composeStruct.ImageBuilds) == 0 {
+			panic("the was a compose with zero image builds, that is forbidden")
```
Comment (msehnout) [inline internal/store/json.go]: `s|the|there|`?

File: internal/store/json.go [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+		blueprintsCommits: newCommitsFromV0(storeStruct.Commits),
+	}
+
+	// Backwards compatibility: fail all builds that are queued or
+	// running. Jobs status is now handled outside of the store
+	// (and the compose). The fields are kept so that previously
+	// succeeded builds still show up correctly.
+	for composeID, compose := range store.composes {
+		if len(compose.ImageBuilds) == 0 {
+			panic("the was a compose with zero image builds, that is forbidden")
```
Comment (msehnout) [inline internal/store/json.go]: same as above

File: internal/store/json.go [hunk 7]
```diff
@@ -239,30 +245,46 @@ func newStoreFromV0(storeStruct storeV0, arch distro.Arch) *Store {
...
+	workspaceStruct := make(workspaceV0)
+	for name, blueprint := range workspace {
+		workspaceStruct[name] = blueprint.DeepCopy()
+	}
+	return workspaceStruct
+}
+
+func newComposeV0(compose Compose) composeV0 {
+	return composeV0{
+		Blueprint: compose.Blueprint, // TODO: deep copy
```
Comment (msehnout) [inline internal/store/json.go:255]: Why is this TODO? :) Is there any problem with calling `compose.Blueprint.DeepCopy()`?

File: internal/test/helpers.go (hunk 4 above)
Comment (teg) [reply internal/test/helpers.go:161]: The idea is that given a triple distro, arch, type the object you get by resolving them should be considered the same, and indeed all the other functions return the same results. I'll add a comment.

File: internal/store/json.go (hunk 5 above)
Comment (teg) [reply internal/store/json.go]: Thanks!

File: internal/store/json.go (hunk 6 above)
Comment (teg) [reply internal/store/json.go]: Thanks!

File: internal/store/json.go (hunk 7 above)
Comment (teg) [reply internal/store/json.go:255]: This looks like an oversight, I'll fix!

File: internal/store/json.go (hunk 6 above)
Comment (teg) [reply internal/store/json.go]: Ah, this was dropped in a later patch. I'll leave the typo in the git history ;)

File: internal/store/json.go (hunk 5 above)
Comment (teg) [reply internal/store/json.go]: This was dropped in a later patch, I'll leave the typo in the git history ;)

**Comment by teg** [comment]
Addressed the reviews, added a few more tests (but more would be great!). Ready to be merged from my point of view.

**Pull request merged**